parlai/mturk/core/socket_manager.py

```diff
@@ -718,3 +717,73 @@ def shutdown(self):
         self.close_all_channels()
         self.keep_running = False
         self._ensure_closed()
+
+
+# TODO extract common functionality from the above and make both classes extend
+# from the base
+class StaticSocketManager(SocketManager):
+    """Version of SocketManager that communicates consistently with the world,
+    but isn't keeping track of the liveliness of the agents that connect as
+    these are single person tasks. Submissions are handled via post rather
+    than served over socket, so it doesn't make sense to.
+    """
+    def channel_thread(self):
+        """Handler thread for monitoring all channels to send things to"""
+        # while the thread is still alive
+        while not self.is_shutdown:
+            for connection_id in self.run.copy():
+                if not self.run[connection_id]:
+                    continue
+                try:
+                    # Make sure the queue still exists
+                    if connection_id not in self.queues:
+                        self.run[connection_id] = False
+                        break
+                    if self.blocking_packets.get(connection_id) is not None:
+                        packet_item = self.blocking_packets[connection_id]
+                        if not self.packet_should_block(packet_item):
+                            self.blocking_packets[connection_id] = None
+                        else:
+                            continue
+                    try:
+                        # Get first item in the queue, check if can send it yet
+                        item = self.queues[connection_id].get(block=False)
+                        t = item[0]
+                        if time.time() < t:
+                            # Put the item back into the queue,
+                            # it's not time to pop yet
+                            self._safe_put(connection_id, item)
+                        else:
+                            # Try to send the packet
+                            packet = item[1]
+                            if not packet:
+                                # This packet was deleted out from under us
+                                continue
+                            if packet.status is not Packet.STATUS_ACK:
+                                # either need to send initial packet
+                                # or resend not-acked packet
+                                self._send_packet(packet, connection_id, t)
+                    except Empty:
+                        pass
+                except BaseException as e:
```

Comment:
why baseexception, why capture e?

Reply:
No particular reason other than inheritance from old code. Should be replaced with Exception. We want to capture though because sockets can have a varied series of errors and if this thread dies the whole HIT dies too. I really should contain the scope of this at some point, but haven't had the bandwidth.